### PR TITLE
Integrate self-hoster runners

### PR DESCRIPTION
**Pull request by pavlovic-ivan**
Hi,

This PR introduces self-hosted runners (called ubuntu-20.04-arm64 in this case) and runs all C# and F# unit tests for netcoreapp3.1, net6.0 and net7.0.

Working example: https://github.com/pavlovic-ivan/ParquetSharp/actions/runs/5819182344

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -215,16 +215,13 @@ jobs:
   test-nuget:
     strategy:
       matrix:
-        os: [ubuntu-20.04, macos-11, macos-12, windows-2022]
+        os: [ubuntu-20.04, macos-11, macos-12, windows-2022, ubuntu-20.04-arm64]
         dotnet: [netcoreapp3.1, net6.0, net7.0]
         arch: [x64]
```
Comment (jgiannuzzi) [inline .github/workflows/ci.yml:220]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -215,16 +215,13 @@ jobs:
...
         include:
         - os: windows-2022
           dotnet: net472
           arch: x64
```
Comment (jgiannuzzi) [inline .github/workflows/ci.yml:224]:
```suggestion
```

**Review by jgiannuzzi — APPROVED** [review]
LGTM 🥇

**Comment by pavlovic-ivan** [comment]
Thank you as well @jgiannuzzi for the all the help!

**Pull request merged**